test/modeltests/testactivitymodel.cpp

```diff
@@ -0,0 +1,58 @@
+
+/*
+ *    This software is in the public domain, furnished "as is", without technical
+ *    support, and with no warranty, express or implied, as to its usefulness for
+ *    any purpose.
+ *
+ */
+
+#include "gui/activitylistmodel.h"
+#include "gui/accountmanager.h"
+
+#include <QTest>
+#include <QAbstractItemModelTester>
+
+namespace OCC {
+
+class TestActivityModel : public QObject
+{
+    Q_OBJECT
+
+private Q_SLOTS:
+    void testInsert()
+    {
+        auto model = new ActivityListModel(this);
+
+        new QAbstractItemModelTester(model, this);
+
+        auto manager = AccountManager::instance();
+
+        auto createAcc = [&] {
+            // don't use the account managher to create the account, it would try to use widgets
```

Comment:
```suggestion
            // don't use the account manager to create the account, it would try to use widgets
```

Reply:
Perhaps the account manager perhaps needs to be extended with such functionality?